components/search/src/configuration_types.rs

```diff
@@ -91,12 +91,71 @@ pub(crate) struct JSONEngineBase {
     pub urls: JSONEngineUrls,
 }
 
+/// Specifies details of possible user environments that the engine or variant
+/// applies to.
+#[derive(Debug, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct JSONVariantEnvironment {
+    /// Indicates that this section applies to all regions and locales. May be
+    /// modified by excludedRegions/excludedLocales.
+    pub all_regions_and_locales: Option<bool>,
+
+    /// A vector of locales that this section should be excluded from. 'default'
+    /// will apply to situations where we have not been able to detect the user's
+    /// locale.
+    pub excluded_locales: Option<Vec<String>>,
```

Comment:
It is indeed simpler, and actually highlights a bit where the search config probably isn't as good as it could be (the all regions & locales boolean treated as tri-state). However, I've got it all working & it does simplify it down, so going without the options.

We might be able to change more of the configuration_types to be option-less, however, I'll only do the environment ones at the moment, since that's what this patch is about.